subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/internal/AbstractCodeQualityPlugin.java

```diff
@@ -94,6 +95,8 @@ protected void createConfigurations() {
         configuration.setVisible(false);
         configuration.setTransitive(true);
         configuration.setDescription("The " + getToolName() + " libraries to be used for this project.");
```

Comment:
Given this is the configuration for the tool libraries to be used for *this* project - should this configuration declare an outgoing variant (be consumable)?
If the use case would be to declare an outgoing variant from the given tool configuration, wouldn't it be clearer to then explicitly extend this configuration when needed and declare an outgoing variant with any wanted attributes?
If the above yields yes/yes - what would be the best use case/test for the attributes of this resolvable and not-consumable configuration?

Reply:
This configuration should have `setCanBeResolved(false)`. But changing that is a breaking change which should follow the regular process for its.